mozjs-sys/build.rs

```diff
@@ -791,22 +816,53 @@ fn decompress_static_lib(archive: &Path, build_dir: &Path) -> Result<(), std::io
     let tar_gz = File::open(archive).unwrap_or({
         let mut workspace_dir = get_cargo_target_dir(build_dir).unwrap().to_path_buf();
         workspace_dir.pop();
-        File::open(workspace_dir.join(archive)).unwrap()
+        File::open(workspace_dir.join(archive))?
     });
     let tar = GzDecoder::new(tar_gz);
     let mut archive = Archive::new(tar);
     archive.unpack(build_dir)?;
     Ok(())
 }
 
-/// Download static library tarball instead of building it from source.
-fn download_static_lib_binaries(archive: &Path, build_dir: &Path) {
-    // Only download the files if build directory doesn't exist.
-    if !build_dir.exists() {
-        // TODO download from https
-        decompress_static_lib(archive, build_dir).expect("Failed to decompress static libs");
+/// Download spidermonkey archive by curl with provided base url. If it's None, it will use
+/// servo/mozjs's release page as base url.
+fn download_archive(base: Option<&str>) -> Result<PathBuf, std::io::Error> {
+    let base = base.unwrap_or("https://github.com/servo/mozjs/releases/download");
+    let version = env::var("CARGO_PKG_VERSION").unwrap();
+    let target = env::var("TARGET").unwrap();
+    let archive_path = PathBuf::from(env::var_os("OUT_DIR").unwrap()).join("libmozjs.tar.gz");
+    if !archive_path.exists() {
+        if !Command::new("curl")
```

Comment:
We should document somewhere that cURL is now a dependency.

Reply:
I added in mozjs's README. I'll also add them to servo's README later.

Reply:
It would be nice to add this for Windows as well.

Reply:
Windows has builtin curl and both cmd and powershell seem working in my local test.